### PR TITLE
devmapper: implement usage

### PR DESCRIPTION
**Pull request by mxpv**
Signed-off-by: Maksym Pavlenko <makpav@amazon.com>

*Issue #, if available:*

*Description of changes:*
This PR implements snapshotter's `Usage` method.
Also adds a few minor fixes and improvements.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: snapshotter/devmapper/snapshotter_test.go [hunk 1]
```diff
@@ -16,61 +16,114 @@ package devmapper
...
-
-			return err
-		}, nil
+		snap := createSnapshotter(t, root)
+		return snap, snap.Close, nil
+	})
+}
+
+func TestSnapshotterUsage(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "devmapper-snapshotter-")
```
Comment (samuelkarp) [inline snapshotter/devmapper/snapshotter_test.go]: This looks like it needs `internal.RequiresRoot(t)`

File: snapshotter/devmapper/snapshotter.go [hunk 2]
```diff
@@ -112,11 +112,32 @@ func (s *Snapshotter) Update(ctx context.Context, info snapshots.Info, fieldpath
...
+	}
+
+	deviceName := s.getDeviceName(id)
+	devicePath := dmsetup.GetFullDevicePath(deviceName)
+	blockDeviceSize, err := dmsetup.BlockDeviceSize(devicePath)
+	if err != nil {
+		return snapshots.Usage{}, err
+	}
+
+	return snapshots.Usage{Size: int64(blockDeviceSize)}, nil
```
Comment (xibz) [inline snapshotter/devmapper/snapshotter.go]: Out of curiosity, what about the `Inodes` field? And out of curiosity why can't you use the `Size` that is returned by `GetInfo`?

Comment (xibz) [reply snapshotter/devmapper/snapshotter.go]: Awesome, would be curious about those benchmarks. I think for me to approve this PR, I would either like to see this implemented in the `GetInfo` portion and have it calculate inodes, and if it turns out to be a huge hit on performance figure something out or have a comment signifying a that inodes are missing from the `Usage` call. May be a good idea to have a TODO about the inode stuff all together.

**Comment by mxpv** [comment]
Will be reworked in separate PR.

**Pull request closed without merging**